azurerm/resource_arm_servicebus_rule.go

```diff
@@ -0,0 +1,354 @@
+
+	if properties := resp.Ruleproperties; properties != nil {
+		d.Set("filter_type", properties.FilterType)
+
+		if properties.Action != nil {
+			d.Set("action", properties.Action.SQLExpression)
+		}
+
+		if properties.SQLFilter != nil {
+			d.Set("sql_filter", properties.SQLFilter.SQLExpression)
+		}
+
+		if properties.CorrelationFilter != nil {
+			if err := d.Set("correlation_filter", flattenAzureRmServiceBusCorrelationFilter(properties.CorrelationFilter)); err != nil {
+				return fmt.Errorf("Error setting `correlation_filter` on Azure ServiceVus Rule (%q): %+v", name, err)
+			}
+		}
+	}
+
+	return nil
+}
+
+func resourceArmServiceBusRuleDelete(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*ArmClient).serviceBusRulesClient
+	ctx := meta.(*ArmClient).StopContext
+
+	id, err := parseAzureResourceID(d.Id())
+	if err != nil {
+		return err
+	}
+	resourceGroup := id.ResourceGroup
+	namespaceName := id.Path["namespaces"]
+	topicName := id.Path["topics"]
+	subscriptionName := id.Path["subscriptions"]
+	name := id.Path["rules"]
+
+	_, err = client.Delete(ctx, resourceGroup, namespaceName, topicName, subscriptionName, name)
+
+	return err
+}
+
+func getAzureRmServiceBusAction(d *schema.ResourceData) *servicebus.Action {
+	if action := d.Get("action").(string); action != "" {
+		serviceBusAction := servicebus.Action{
+			SQLExpression: &action,
+		}
+		return &serviceBusAction
+	}
+	return nil
+}
+
+func getAzureRmServiceBusSQLFilter(d *schema.ResourceData) *servicebus.SQLFilter {
+	if sqlFilter := d.Get("sql_filter").(string); sqlFilter != "" {
+		serviceBusSQLFilter := servicebus.SQLFilter{
+			SQLExpression: &sqlFilter,
+		}
+		return &serviceBusSQLFilter
+	}
+	return nil
+}
+
+func getAzureRmServiceBusCorrelationFilter(d *schema.ResourceData) *servicebus.CorrelationFilter {
+	if config := d.Get("correlation_filter").([]interface{}); len(config) > 0 {
+		filter := config[0].(map[string]interface{})
+		correlationFilter := servicebus.CorrelationFilter{}
+
+		if correlationID := filter["correlation_id"].(string); correlationID != "" {
+			correlationFilter.CorrelationID = &correlationID
+		}
+
+		if messageID := filter["message_id"].(string); messageID != "" {
+			correlationFilter.MessageID = &messageID
+		}
+
+		if to := filter["to"].(string); to != "" {
+			correlationFilter.To = &to
+		}
+
+		if replyTo := filter["reply_to"].(string); replyTo != "" {
+			correlationFilter.ReplyTo = &replyTo
+		}
+
+		if label := filter["label"].(string); label != "" {
+			correlationFilter.Label = &label
+		}
+
+		if sessionID := filter["session_id"].(string); sessionID != "" {
+			correlationFilter.SessionID = &sessionID
+		}
+
+		if replyToSessionID := filter["reply_to_session_id"].(string); replyToSessionID != "" {
+			correlationFilter.ReplyToSessionID = &replyToSessionID
+		}
+
+		if contentType := filter["content_type"].(string); contentType != "" {
+			correlationFilter.ContentType = &contentType
+		}
+
+		return &correlationFilter
```

Comment:
based on this comment: https://github.com/terraform-providers/terraform-provider-azurerm/pull/1124/files#diff-5ae250cddcb7acd8b53f87d1465954bfR168 - is it worth raising an error if no properties have been set?